### DISCUSSION TITLE
DNNL EP how to access data in TensorProto struct

### DISCUSSION THREAD

**Discussion by zhaoyang-intel** [microsoft/onnxruntime#7600] [Other Q&A] [answered]
I'm trying to save the model initializers' data in DNNL EP and the TensorProto struct in provider_interfaces.h doesn't have .float_data() exposed. Is there a workaround for this?

**Comment by jywu-mysoft** [comment, marked as answer]
AFAIK , existing interfaces were added based on provider usage at the time, so there may be missing interfaces.
@ryanunderhill , what is the recommendation for developers here?

**Reply by RyanUnderhill** [reply]
Yeah, it was impractical to add every method just due to the nature of how we're routing the stuff across to a shared library. It's fairly straightforward to extend the interface. I recommend looking at the history of the provider_interfaces.h file as an example of what is affected by adding a new function.

If you need any help during it, just let me know.

**Reply by zhaoyang-intel** [reply]
Thank you! I will follow your suggestion to add the needed methods.
